src/addrman.h

```diff
@@ -318,8 +329,15 @@ friend class CAddrManTest;
     {
         LOCK(cs);
 
-        unsigned char nVersion = 2;
-        s << nVersion;
+        const auto stream_version_orig = s.GetVersion();
+
+        if (m_format >= Format::V3) {
+            // Add ADDRV2_FORMAT to the version so that the CNetAddr and CAddress
+            // serialize methods produce an address in v2 format.
+            s.SetVersion(stream_version_orig | ADDRV2_FORMAT);
+        }
+
+        s << (uint8_t)m_format;
```

Comment:
e025121c1ca4b86296c6b36386786522bd79a6e1

```suggestion
        s << static_cast<uint8_t>(m_format);
```

Reply:
Done

Reply:
Per https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast and *C++ Programming Language 4th Edition*, section 11.5, it might be good to update the next line as well if you retouch.
```diff
         s << static_cast<uint8_t>(m_format);
-        s << ((unsigned char)32);
+        s << static_cast<unsigned char>(32);
```

Reply:
That line is unmodified by the patch. I try to avoid sneaking unrelated improvements to surrounding code in order to ease reviewers and future history observers.

Reply:
That's fair.